### DISCUSSION TITLE
「パターンをWeb画面で見れる」に関するスレ

### DISCUSSION THREAD

**Discussion by msntts** [msntts/PatternLanguageEditor#4] [General]
開発環境も整ったので、いよいよ開発に取り掛かろうと思います。
とはいえ、二人ともReact/Reduxって何・・・っていう状態だと思うので、ここでやり取りしながら進めていきましょう。
やり取りは何でもいいです。ここのHPのこれが使えそう～とかそういうものでも大丈夫です。
ブランチ名は`display-pattern`にしようと思います。
最終的にsquash mergeするので、ちょい変え→pushしてコードを共有しながら進めていけたらいいと思います。

得られた知見はnoteに反映していきましょう✊

**Comment by msntts** [comment]
とりあえず、パターンの表示だけ作ってみた
[コミット履歴](https://github.com/msntts/PatternLanguageEditor/commit/088c98e82c4fda459130ec415ddb9e9222e11b5f
)
git pull して`npm install`(いらないかも) -> `npm start`すると味気ない画面が出ると思います。
この後は、patternのモデルをつくる→App.tsx(?)でPattern.tsとつなぐみたいな感じ？ちょっとここがまだあやしい。。

**Reply by kinakomochigit** [reply]
画面表示成功しました。

**Comment by kinakomochigit** [comment]
「パターンのモデルを作成する」について初歩的な質問です。  
どのような状態になることが完了なのかイメージの確認と、実装に関する質問です。

1. 完了イメージの確認
`npm start`をすると[kintone SIGNPOST](https://kintone.cybozu.co.jp/kintone-signpost/pattern/0-00.html)のように画像と各項目の文章が現れる。という理解であってますか？


2. 実装について
「patternのモデルをつくる」というのは「App.tsx(?)でPattern.tsxとつなぐ」というのは前のディスカッションで書かれていあるので理解しました。
そのうえでつなぐというのはApp.tsx、Pattern.tsx間で画像のパスの受け渡し方を考える感じでしょうか？
私の中ではPattern.tsxがApp.tsxで利用されている状態と認識しています。単に`imgPath=""`に画像のパスを入れればよいというわけではない感じでしょうか？
現在React理解中でコンポーネントとかまだあやふやなので、キーワードをいただければ、そのあたり見ておきます！

**Reply by msntts** [reply]
```
完了イメージの確認
npm startをするとkintone SIGNPOSTのように画像と各項目の文章が現れる。という理解であってますか？
```
そうです。ゆくゆくは初期画面に違うものが出るかもしれないですが、まずは、npm startで画像と各項目の文章(装飾もないもの)がでればOKです

```
そのうえでつなぐというのはApp.tsx、Pattern.tsx間で画像のパスの受け渡し方を考える感じでしょうか？
```
この辺話が難しくて(というか勘違いしてた)、tsxはあくまでもviewの話なので、モデルではないです。
今回のタスクはview(tsx)とモデルをつなぐ、要はデータモデルに書かれているものを画面に表示する。　というゴールです
今のブランチは、viewのコードしかないので、データモデルのフォルダから作る必要があります。
こっちのほうでちょっとかじっているものがあるので、あとでpushしておきます。(ただし、全然動いてない)

```
現在React理解中でコンポーネントとかまだあやふやなので、キーワードをいただければ、そのあたり見ておきます！
```
コンポーネントとかの話は、Reactというか、実装のパターンみたいなものかなぁと思っています。
なのでちょっとブログなりに書いたほうがいいかもしれないですね。
モデルに関してはRedux toolkitっていうのをつかったらいいなーと思っていて、今はSliceについて勉強中です。
これも仕掛だけどブログにかいたほうがいいすね

**Reply by msntts** [reply]
↑の文章Pattern.tsx->Pattern.tsに修正しました！

**Reply by msntts** [reply]
pushしました✌ まだまだ作りかけだけど、Viewとのつなぎはこんなイメージ

**Reply by msntts** [reply]
いったん、モデルとつなぐコードをpushしてみました。(noteのほうもrev.0.2で更新)
ここで、一回歩幅を合わせようと思います。(自分がワー！と作ってる状態なので)
どこかで一回、コードの説明会をしましょう✊

**Reply by msntts** [reply]
あと、パッケージを追加したので、npm installをお願いします🥺

**Reply by kinakomochigit** [reply]
コード説明会よろしくお願いいたします！

**Reply by kinakomochigit** [reply]
21日21時〜
26日の10時以降可能です
